modules/standard/UtilReplicatedVar.chpl

```diff
@@ -59,38 +63,60 @@ How to use replicated variables:
     // access directly a remote copy on the locale 'remoteLoc' (read or write)
     ... rcRemote(myRepVar, remoteLoc) ...
 
-Advanced: replicate over a subset of locales 'myLocales'.
-// 'myLocales' must be "consistent", as defined for ReplicatedDist.
-Use it as above, except modify the variable declarations as follows:
+For more examples, see :proc:`rcExample` and :proc:`rcExampleOverLocales`
+in the module source code.
+
+----------------------------------------------
+Advanced: replicating over a subset of locales
+----------------------------------------------
+
+To replicate a variable over a subset of locales, say ``myLocales``,
+modify the above variable declarations as follows::
 
     var myRepVar: [rcDomainBase dmapped ReplicatedDist(myLocales,
                      "over which to replicate 'myRepVar'")] MyType;
     var collected: [myLocales.domain] MyType;
 
-See also rcExample() and rcExampleOverLocales() in this file.
-*/
+``myLocales`` must be "consistent", as defined for ReplicatedDist.
+That is, for each ``ix`` in ``myLocales.domain``,
+``myLocales[ix]`` must equal ``Locales[ix]``.
 
-/* */
+Tip: if the domain of the desired array of locales cannot be described
+as a rectangular domain (which could be strided, multi-dimensional,
+and/or sparse), make that array's domain associative over int.
+*/
+module UtilReplicatedVar {
 
 use ReplicatedDist;
 
+pragma "no doc"
 const rcDomainIx   = 1;
+/* Use this domain when replicating over a subset of locales,
+   as shown above. */
 const rcDomainBase = {rcDomainIx..rcDomainIx};
+pragma "no doc"
 const rcLocales    = Locales;
 const rcDomainMap  = new ReplicatedDist(rcLocales);
+/* Use this domain to declare a user-level replicated variable,
+   as shown above. */
 const rcDomain     = rcDomainBase dmapped new dmap(rcDomainMap);
+pragma "no doc" // todo - remove this?  our examples use LocaleSpace instead
 const rcCollectDomaim = rcLocales.domain;
+pragma "no doc"
 param _rcErr1 = " must be 'rcDomain' or 'rcDomainBase dmapped ReplicatedDist(an array of locales)'";
 
+pragma "no doc"
 proc _rcTargetLocalesHelper(replicatedVar: [?D])
   where replicatedVar._value.type: ReplicatedArr
 {
   return replicatedVar._value.dom.dist.targetLocales;
 }
 
+pragma "no doc" // documented with the following entry
 proc rcReplicate(replicatedVar: [?D] ?MYTYPE, valToReplicate: MYTYPE): void
 { compilerError("the domain of first argument to rcReplicate()", _rcErr1); }
 
+/* Assign `valToReplicate` to copies of `replicatedVar` on all locales. */
```

Comment:
Consider adding more detail here.  What is `valToReplicate` intended to represent?  What about `replicatedVar`?

Remember, someone may reach the documentation for this function without going through your module description - if they're looking at someone else's code, or another module's documentation referenced this function, for instance.  Documentation for each function should give a full picture of what that function does itself, while the module level comment serves to put each of these distinct pieces in context.